src/bootstrap/lib.rs

```diff
@@ -668,6 +668,7 @@ impl Build {
             job::setup(self);
         }
 
+        format::format(&builder::Builder::new(&self), true, &[]);
         self.maybe_update_submodules();
```

Comment:
This isn't right, it will unconditionally *run* rustfmt whenever x.py is run. We should only *download* rustfmt. Take a look at what `format` is doing internally.